### PR TITLE
Jetpack License Portal: Add license attachment page and required dependencies

### PR DESCRIPTION
**Pull request by rcoll**
#### Changes proposed in this Pull Request

* This PR adds the license attachment screen to the licensing portal, allowing newly issued licenses to be attached without leaving the portal.

#### Testing instructions

* Issue a new license and ensure that the license attachment page loads immediately after.
* Attempt to attach the license to a valid/connected/owned Jetpack site.
* License should attach and reflect as such in the licenses list.

### CODE REVIEW

File: client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx [hunk 1]
```diff
@@ -149,14 +149,12 @@ export default function LicensePreview( {
...
-							text={ licenseKey }
-							className="license-preview__copy-license-key"
+						<Button
 							compact
-							onCopy={ onCopyLicense }
+							href={ addQueryArgs( { key: licenseKey }, '/partner-portal/assign-license' ) }
 						>
-							{ translate( 'Copy License' ) }
-						</ClipboardButton>
+							{ translate( 'Assign License' ) }
```
Comment (a8ci18n) [inline client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx:167]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **4** times:
`translate( 'Unable to assign license.' )` *ES Score: 7*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-license-portal-license-attachment)

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+			return (
+				<Card key={ site.ID } className="assign-license-form__site-card">
+					<FormRadio
+						className="assign-license-form__site-card-radio"
+						label=""
+						name="site_select"
+						onClick={ () => onSelectSite( site.ID ) }
+					/>
+					{ site.domain }
+				</Card>
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: I noticed a few things when interacting with the sites list:

1. `site.domain` isn't set as the `FormRadio` label property
2. `form-radio__label` isn't a `<label>` but rather a `<span>`, so even after we address bullet 1, we still have to do some more work
3. The components padding is defined on `.assign-license-form__site-card` instead of the label - if we move it to the label, the entire box should make it clickable
4. There's no hover state (neglecting hover state is sometimes a conscious design decision when designing mobile first, so I don't know if that's the case here, but most other interactive components on the page has it)

Screenshot 2022-02-09 at 12 09 38

Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: It looks like we're using the default browser styling instead of the "Jetpack checkmark" from the design. Could the checkbox used in `<LicenseProductCard/>` be generalised/copied to this one as well?

Screenshot 2022-02-09 at 12 22 49

---

The radio and label isn't flush with each other

Screenshot 2022-02-09 at 12 20 16

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+	} );
+
+	const onAssignLicense = useCallback( () => {
+		setIsSubmitting( true );
+		dispatch( infoNotice( translate( 'Assigning license…' ) ) );
+		dispatch(
+			recordTracksEvent( 'calypso_partner_portal_assign_license_submit', {
+				licenseKey,
+				selectedSite,
+			} )
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: I get the following console error for this event:
```
Tracks: Event `calypso_partner_portal_assign_license_submit` will be rejected because property name `licenseKey` does not match /^[a-z_][a-z0-9_]*$/. Please use a compliant property name.
```

I believe it's because `recordTracksEvent` expects property names in the properties object - e.g.:
```js
recordTracksEvent( 'calypso_partner_portal_assign_license_submit', {
    /* Other places in Calypso use snake_case for the property names */
    license_key: licenseKey,
    selected_site: selectedSite,
} )
```

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+			);
+		}
+	} );
+
+	const onSearch = ( query: any ) => setFilter( query );
+
+	const assignLicense = useAssignLicenseMutation( {
+		onSuccess: ( licenseKey: any ) => {
+			setIsSubmitting( false );
+			page.redirect( addQueryArgs( { highlight: licenseKey }, '/partner-portal/licenses' ) );
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: The `highlight` parameter value seems to be incorrectly passed along (I see `/partner-portal/licenses?highlight=%5Bobject+Object%5D` after being redirected).

If I remember the key correctly, it should work with:

```js
page.redirect( addQueryArgs( { highlight: licenseKey.license_key }, '/partner-portal/licenses' ) );
```

---


Maybe we should rename the return variable to something else to avoid confusion? Perhaps `license` instead of `licenseKey`?

File: client/jetpack-cloud/sections/partner-portal/controller.tsx [hunk 5]
```diff
@@ -93,6 +95,17 @@ export function issueLicenseContext( context: PageJS.Context, next: () => void )
...
 }
 
+export function assignLicenseContext( context: PageJS.Context, next: () => void ): void {
+	const state = context.store.getState();
+	const sites = getSites( state );
+
+	context.header = <Header />;
+	context.secondary = <PartnerPortalSidebar path={ context.path } />;
+	context.primary = <AssignLicense sites={ sites } />;
+	context.footer = <JetpackComFooter />;
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/controller.tsx]: Not for now - this is 100% 👍 -  but in the future we could look into creating a wrapper function for all our partner pages so we don't have to keep defining (DRY approach)

```js
context.header = <Header />; // <-- The same on all license portal pages
context.secondary = <PartnerPortalSidebar path={ context.path } />; // <-- The same on all license portal pages
context.primary = <SomethingDifferent />;
context.footer = <JetpackComFooter />; // <-- The same on all license portal pages
```

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import { useMutation, UseMutationOptions, UseMutationResult } from 'react-query';
+import { wpcomJetpackLicensing as wpcomJpl } from 'calypso/lib/wp';
+import { APILicenseAssign } from 'calypso/state/partner-portal/types';
+
+interface MutationAssignLicenseVariables {
+	licenseKey: string;
+	selectedSite: any;
```
Comment (kallehauge) [inline client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: Could we increase the specificity of `selectedSite` by defining it as a number? AFAIK it should always be a blog ID

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+		},
+		onError: ( error: Error ) => {
+			setIsSubmitting( false );
+			dispatch( errorNotice( error.message ) );
+		},
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:50]: **Remove initial notification/state**

AFAIK it's possible to pass an ID when dispatching a notice, so it can be programatically removed later (see `client/state/notices/actions.ts::removeNotice`).
Should we perhaps do that here, so we remove the initial `Assigning license...` message on errors?

Screenshot 2022-02-09 at 13 06 10

---

**Design differences**

Error messages appears differently than in the design - this is probably a way to limit the scope since we already have notifications, but just wanted to highlight it.

Screenshot 2022-02-09 at 13 05 58

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+				</p>
+				<div className="assign-license-form__controls">
+					<Button
+						borderless
+						onClick={ onAssignLater }
+						disabled={ isSubmitting }
+						className="assign-license-form__assign-later"
+					>
+						{ translate( 'Assign later' ) }
+					</Button>
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:81]: Is this a clever way to reduce the scope of the `or pick a website later` link, so we don't have to detect the context we enter the page from? 😄 

Screenshot 2022-02-09 at 13 24 25

File: client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx [hunk 9]
```diff
@@ -101,12 +101,18 @@ export default function LicensePreview( {
...
 							<span className="license-preview__tag license-preview__tag--is-just-issued">
 								<Gridicon icon="checkmark-circle" size={ 18 } />
 								{ translate( 'Just issued' ) }
 							</span>
 						) }
+						{ justAssigned && (
+							<span className="license-preview__tag license-preview__tag--is-assigned">
+								<Gridicon icon="checkmark-circle" size={ 18 } />
+								{ translate( 'Successfully assigned' ) }
+							</span>
```
Comment (kallehauge) [inline client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx:119]: This is currently not showing up I'm assuming it will after the `highlight` query param feedback has been looked at 😄

**Review by kallehauge — COMMENTED** [review]
Generally it looks really good! I've added a few inline comments with what I've found.
I've not tried to decide what seemed like triming scope, but rather just reporting what I saw, so feel free to skip things 😄

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+						name="site_select"
+						onClick={ () => onSelectSite( site.ID ) }
+					/>
+					{ site.domain }
+				</Card>
+			);
+		}
+	} );
+
+	const onSearch = ( query: any ) => setFilter( query );
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: We can skip this and pass setFilter directly as the event callback. That will also avoid some unnecessary renders as the value won't change.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+import useAssignLicenseMutation from 'calypso/state/partner-portal/licenses/hooks/use-assign-license-mutation';
+import './style.scss';
+
+export default function AssignLicenseForm( { sites }: any ): ReactElement {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const [ filter, setFilter ] = useState( false );
+	const [ selectedSite, setSelectedSite ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const licenseKey = getQueryArg( window.location.href, 'key' ) as string;
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:21]: Not that important: We should follow the convention of page context functions passing query parameters down as component parameters (e.g. `licensesContext`) rather than components getting query parameters on their own, so components are "more" reusable.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+				<div className="assign-license-form__controls">
+					<Button
+						borderless
+						onClick={ onAssignLater }
+						disabled={ isSubmitting }
+						className="assign-license-form__assign-later"
+					>
+						{ translate( 'Assign later' ) }
+					</Button>
+					<Button primary onClick={ onAssignLicense } disabled={ isSubmitting }>
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: We can use the `busy` prop instead of `disabled` here which will disable the button and apply a suitable animation.
By using `busy` we can also drop the "Assigning license...` notice entirely.

File: client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx [hunk 13]
```diff
@@ -50,7 +50,7 @@ export default function LicensePreview( {
 		domain && [ LicenseState.Attached, LicenseState.Revoked ].indexOf( licenseState ) !== -1;
 	const justIssued =
 		moment.utc( issuedAt, 'YYYY-MM-DD HH:mm:ss' ) > moment.utc().subtract( 1, 'minute' );
-
+	const justAssigned = justIssued && 'attached' === licenseState;
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx]: - Let's use the enum we already have: LicenseState.Attached
- justIssued cannot be reused here as a license can be assigned days after it has been issued so we should adjust the behavior to match but test the attachedAt date rather than the issuedAt date:
```
const aMinuteAgo = moment.utc().subtract( 1, 'minute' );
const justIssued = moment.utc( issuedAt, 'YYYY-MM-DD HH:mm:ss' ) > aMinuteAgo;
const justAssigned =
    moment.utc( attachedAt, 'YYYY-MM-DD HH:mm:ss' ) > aMinuteAgo &&
    LicenseState.Attached === licenseState;
```

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+	useEffect( () => {
+		const layoutClass = 'layout__content--partner-portal-assign-license';
+		const content = document.getElementById( 'content' );
+
+		if ( content ) {
+			content.classList.add( layoutClass );
+
+			return () => content.classList.remove( layoutClass );
+		}
+	}, [] );
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx:21]: We should move this to a hook so we don't keep duplicating it for other similar layouts.

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+	return (
+		<Main wideLayout className="assign-license">
+			<DocumentHead title={ translate( 'Assign your new License' ) } />
+			<SidebarNavigation />
+			<CardHeading size={ 36 }>{ translate( 'Assign your new License' ) }</CardHeading>
```
Comment (atanas-dev) [inline client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx]: The word "new" in `your new License` does not make sense when assigning an old license. I don't have strong opinions on whether we should keep it as-is or just drop the word in both cases for simplicity.

File: client/state/partner-portal/types.ts [hunk 16]
```diff
@@ -72,6 +72,11 @@ export interface APILicense {
 	revoked_at: string | null;
 }
 
+export interface APILicenseAssign {
+	ok: boolean;
+	error: string;
+}
+
```
Comment (atanas-dev) [inline client/state/partner-portal/types.ts]: This is unnecessary - the assign endpoint responds with a license which is covered by the `APILicense` interface.

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+		apiNamespace: 'wpcom/v2',
+		path: `/jetpack-licensing/license/${ licenseKey }/site`,
+		body: { site: selectedSite },
+	} );
+}
+
+export default function useAssignLicenseMutation< TContext = unknown >(
+	options?: UseMutationOptions< APILicenseAssign, Error, MutationAssignLicenseVariables, TContext >
+): UseMutationResult< APILicenseAssign, Error, MutationAssignLicenseVariables, TContext > {
+	return useMutation< APILicenseAssign, Error, MutationAssignLicenseVariables, TContext >(
```
Comment (atanas-dev) [inline client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: `APILicenseAssign` -> `APILicense` (see client/state/partner-portal/types.ts comment for context)

**Review by atanas-dev — COMMENTED** [review]
Punt any feedback items to a separate task as you see fit :)

---

The <990px screen sizes could use some love:
Screen Shot 2022-02-09 at 16 27 54

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 11 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:21]: Agreed, but let's address this in a different PR.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 7 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:50]: I've removed the infoNotice in lieu of a different solution, so this is no longer an issue. As for the missing banner, we'll address this down the road, since right now we don't have a great way to map errors into the UI.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 3 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: Good catchh. Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/f5f67dfdf7b9a134f0c53c713806454d7af5e82b

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 8 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:81]: That link is present at the top of the page now in the latest (i3) designs. As for page context - yes, it's an issue, but not a blocker for now.

File: client/jetpack-cloud/sections/partner-portal/controller.tsx (hunk 5 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/controller.tsx]: Agreed. I think we can do a short sprint to clean up some of these patterns.

File: client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx (hunk 9 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/license-preview/index.tsx:119]: Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx (hunk 14 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx:21]: Agreed. As discussed above, we need to do a short sprint to clean up some of these patterns. For now, this isn't a blocker IMO.

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx (hunk 15 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx]: I think this should change based on page context, which we currently lack. I'm in favor of just dropping the word "new" for now. Thoughts?

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts (hunk 6 above)
Comment (rcoll) [reply client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: I think there are cases where `selectedSite` is a site slug, no?

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx (hunk 15 above)
Comment (kallehauge) [reply client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx]: > I'm in favor of just dropping the word "new" for now. Thoughts?

I'm onboard with this decision 👍

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts (hunk 6 above)
Comment (kallehauge) [reply client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: >I think there are cases where selectedSite is a site slug, no?

The endpoint we call [will accept a site slug, full url or blog id](https://github.com/Automattic/jetpack-licensing-api/blob/master/spec.yml#L151-L166).
IMO the site url arguments are mostly for partners who call the API directly and aren't aware of blog ids (we cannot expect partners to be aware of blog ids, so we support multiple types of arguments).

> AFAIK it should always be a blog ID

The way this PR uses the mutation, we should always receive a blog ID.
Theoretically, the blog id is the most specific site identifier we have (think e.g. IDC) and since we're always aware of the blog ID in the licensing portal, I feel like we should enforce that.
It's an opinionated argument, so we can **_definitely_** leave it **_as-is_** since it works, but if we forcing it to be a number, it could help catch errors in the code whereas the current interface will allow anything (strings, numbers, objects, bools, null, etc) and then the endpoint will have to tell us if it looks OK 😄

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 2 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: So I added the label text in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966 but the text still is not clickable. I've also added an additional Asana task to work on this, as it could quickly become out of scope for this PR.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 4 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: Yup. Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 2 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: I've added an additional initiative to unify this component. For now, I don't consider it to be a blocker here, and it could quickly become too complex for this PR. Thoughts?

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 10 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 12 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: Yup, good idea. Done in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.

File: client/state/partner-portal/types.ts (hunk 16 above)
Comment (rcoll) [reply client/state/partner-portal/types.ts]: Good catch. Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.

File: client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx (hunk 15 above)
Comment (rcoll) [reply client/jetpack-cloud/sections/partner-portal/primary/assign-license/index.tsx]: Updated in https://github.com/Automattic/wp-calypso/pull/60903/commits/ef61a9a076788b5cbe6fb038353f67e0a8cb353b

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts (hunk 6 above)
Comment (rcoll) [reply client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: Fair enough. Updated in https://github.com/Automattic/wp-calypso/pull/60903/commits/c71fdab1a09f06f70099c950ac8b882a0a32db14!

File: client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts (hunk 17 above)
Comment (rcoll) [reply client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts]: Good catch. Fixed in https://github.com/Automattic/wp-calypso/pull/60903/commits/9035c9ea6329895054bafb6d41f11392d381d966.

File: client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx (hunk 2 above)
Comment (kallehauge) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: TL;DR: I'm OK with not addressing it in this PR since there's a separate task for it 👍 

---

It's a bit of a pet peeve of mine, so I'm a extra subjective here 😄  I'm a strong advocate for the text to be clickable since that's what people naturally clicks/draws their attention on (_"It's THAT site (read: text) I want to assign it to"_) and an old report I read had it as one of the top bug reports across web development projects.
The report is close to 10 years ago by now and people have become more used to "quirks" on the internet, so I don't mind it being live for a while considering out target audience is agencies as well, but IMO the separate task should be addressed sooner than later considering it's probably a very wanted feature by the agencies we already have using the platform.

_TL;DR of the report:  "Non devs" would not think to click the radio `O`, so they ended up reporting the functionality to not work at all - not just a bad user experience._

Comment (kallehauge) [reply client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx]: Sounds good - the alignment isn't interrupting the user experience either, so this is definitely not a blocker but an enhancement.

**Review by kallehauge — APPROVED** [review]
It tests well for me and all feedback has been addressed, so I'll give it a 👍 
Ship at will @rcoll 🚢 

---

`Unit tests (Web app)` is failing - I just queued it to re-run, but I expect it to pass.

**Review by kallehauge — CHANGES_REQUESTED** [review]
Sorry about that one - I was a bit trigger happy. There's two errors reported in TC 😄 

**Error 1**

This one is good enough and should be addressed (there's a technical chance that we pass the default state `false` to `mutationAssignLicense`):

```
client/jetpack-cloud/sections/partner-portal/assign-license-form/index.tsx:61`: Type 'boolean' is not assignable to type 'number'.
```

**Error 2**

This one is a bit of a mystery to me. I dno if this is a ..eeeh.. is it called "false negative", like "false positive" ? 😄 

```
client/jetpack-cloud/sections/partner-portal/controller.tsx:98: Cannot find name 'JetpackComFooter'.
```

**Review by kallehauge — APPROVED** [review]
Tests are now completing successfully (thanks for fixing it @rcoll). This LGTM 👍

**Comment by heydemoura** [comment]
Stuff that I looked that was a bit awkward was already commented on by others;

Trying it out I got a message saying `Insufficient permissions to attach license to remote site.` Might be something on my partner key.
image

- Spacing for the site list seems uneven, whenever we add the green checkmark icons might look better.
image

**Comment by kallehauge** [comment]
> Trying it out I got a message saying `Insufficient permissions to attach license to remote site.` Might be something on my partner key.

Ah! Yes! It's because you have an internal key 😄  Agencies has some special condition where, as long as they are connected to the site, they're allowed to attach to it.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7139061

Thank you @rcoll for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7139061).
